src/WebSdk/Web/Sdk/Sdk.targets

```diff
@@ -22,6 +22,8 @@ Copyright (c) .NET Foundation. All rights reserved.
       Microsoft.AspNetCore.App. This needs to happen after the .NET SDK has evaluated TFMs.
      -->
     <AddRazorSupportForMvc Condition="'$(DisableImplicitFrameworkReferences)' != 'true' And '$(AddRazorSupportForMvc)' == '' And '$(TargetFrameworkIdentifier)' == '.NETCoreApp' And '$(_TargetFrameworkVersionWithoutV)' >= '3.0'">true</AddRazorSupportForMvc>
+
+    <SuppressTrimAnalysisWarnings Condition="'$(SuppressTrimAnalysisWarnings)' == ''">true</SuppressTrimAnalysisWarnings>
```

Comment:
Doesn't this get set imported after `Microsoft.NET.ILLink.targets`? Meaning by this time, `SuppressTrimAnalysisWarnings` will never be empty because `Microsoft.NET.ILLink.targets` has already set it to false?

I think this should probably move to a `.props` file.

Reply:
Oh, I see you have it in both places. I think this one should be reverted.

---
In reply to: [595273084](https://github.com/dotnet/sdk/pull/16327#discussion_r595273084) [](ancestors = 595273084)

Reply:
The intention of this change was to suppress warnings for aspnet apps. From what I could tell, this isn't imported in blazorwasm projects - and blazorserver projects don't import the BlazorWebAssembly.Current.props. It would be great to have someone double-check this, but this is the SDK dependency graph I came up with:

- .NET
- Web.ProjectSystem
- Publish
- Razor -> .NET
- Worker -> .NET, Web.ProjectSystem
- Web -> .NET, Web.ProjectSystem, Razor, Publish
- BlazorWebassembly -> .NET, Web.ProjectSystem, Razor, Publish

I wonder if we should also suppress warnings for the Worker SDK.

One thing I wasn't sure about was whether it's better to set the web sdk defaults in props or in targets. The .NET SDK defaults are only set in targets, when `PublishTrimmed` is set. But the BlazorWebAssembly.Current.props do it early (since they also set PublishTrimmed early).

Reply:
> The intention of this change was to suppress warnings for aspnet apps. 

Ah, I missed that. I thought this was for Blazor WASM.

> One thing I wasn't sure about was whether it's better to set the web sdk defaults in props or in targets.

This is always a tricky decision. In general, when defaulting properties in a higher-layer, the earlier you can do it the better. This is because the higher the layer you are in, the more information you know and have a better idea at what the value should be.

A higher layer's .props file gets imported before a lower layer's. And a higher layer's .targets file gets imported after a lower layer's. This allows for the higher layer to default properties before the lower layer. And it allows the higher layer to overwrite things after the lower layer has set them.

The issue here is that Microsoft.NET.ILLink.targets is going to default it first. And by the time this setting gets executed, it will already be defaulted and the Condition will no longer hold.

Reply:
Moved this to the Web SDK props for now which should fix the ordering issue you pointed out (I'd forgotten that ILLink targets are part of the .NET SDK targets, not the Publish SDK targets).